test/unit/db/repos/accounts.js

```diff
@@ -549,6 +549,41 @@ describe('db', () => {
 						});
 				});
 			});
+
+			it('should execute all queries in one database transaction', done => {
+				const account = accountFixtures.Account();
+
+				db.$config.options.query = function(event) {
+					if (!(event.ctx && event.ctx.isTX && event.ctx.txLevel === 0)) {
+						return done(
+							`Some query executed outside transaction context: ${event.query}`,
+							event
+						);
+					}
```

Comment:
Can we use expectations here as well. 

```
expect(event.ctx).to.no.null;
expect(event.ctx.isTx).to.be.true;
expect(event.ctx.txLevel).to.be.eql(0);
done();
```

Above will work but not sure how can we send custom error message in case of failure.